hax/hax/ha/event/event.py

```diff
@@ -16,11 +16,45 @@
 # opensource@seagate.com or cortx-questions@seagate.com.
 
 import logging
+import json
+from hax.ha.const import HEALTH_EVENT_SOURCES, NOT_DEFINED
+from hax.util import ConsulUtil
+from cortx.utils.event_framework.health import HealthAttr
 
 
 class HaEvent():
-    def __init__(self):
+    def __init__(self,
+                 source: str,
+                 cluster_id: str,
+                 site_id: str,
+                 rack_id: str,
+                 storageset_id: str,
+                 node_id: str,
+                 resource_type: str,
+                 resource_id: str,
+                 resource_status: str,
+                 specific_info):
         logging.debug('Inside HaEvent')
+        self.payload = {
+            HealthAttr.SOURCE.value: HEALTH_EVENT_SOURCES.HARE.value,
+            HealthAttr.CLUSTER_ID.value: NOT_DEFINED,
+            HealthAttr.SITE_ID.value: NOT_DEFINED,
+            HealthAttr.RACK_ID.value: NOT_DEFINED,
+            HealthAttr.STORAGESET_ID.value: NOT_DEFINED,
+            HealthAttr.NODE_ID.value: node_id,
+            HealthAttr.RESOURCE_TYPE.value: resource_type,
+            HealthAttr.RESOURCE_ID.value: resource_id,
+            HealthAttr.RESOURCE_STATUS.value: resource_status,
+            HealthAttr.SPECIFIC_INFO.value: specific_info
+            }
 
-    def create_event(self, node_id, node_name, health_status):
+    def send(self, util):
         raise NotImplementedError()
+
+    def get_subscribers(self, consul_util: ConsulUtil,
```

Comment:
Additionally, we can also maintain an in-memory cache of subscriptions, whenever a new subscription is added, we update it in-memory and then update the persistent copy. Every time hax restarts this in-memory copy is then loaded in-memory and used for later purposes. As we do not expect subscriptions to be modified too frequently, this will help us optimize the consul kv calls.

Reply:
Done